RobotsAndPencils/slack-salesforce-bot
# Bot that polls PubSub and posts any messages received.

#### What's this PR do?
Adds in config necessary to connect to Google Compute Engine with token required to access PubSub for the slack-salesforce project.  Also, utilizes simple client to pull messages from a subscription to opportunity_changes_sub, and then messages the received data out to any channels the bot is a member of.  Also vendor all the required dependencies.

#### Where should the reviewer start?
main.go, pubsub_events.go, config.go

#### How should this be manually tested?
`gb build all`
`source setenv.sh`
`./bin/bot`

Then login to the slack-salesforce project and publish a message on the opportunity_changes topic:
https://console.cloud.google.com/cloudpubsub/topicList?project=slack-salesforce

Then verify that message is relayed via the bot into the channel its' part of.

#### Any background context you want to provide?
Realized that the appropriate name of PubSubEvents should really be OpportunityChangedEvent... will refactor in an upcoming PR.

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
![image](https://cloud.githubusercontent.com/assets/1517743/13159997/32f1bf70-d652-11e5-8d12-089d5c8db9e3.png)

#### Does the change have adequate test coverage ?
No... trying to get a quick prototype up and running for near term meeting.
